gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/CleanupService.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.services;
+
+import com.netflix.spinnaker.gate.services.commands.HystrixFactory;
+import com.netflix.spinnaker.gate.services.internal.SwabbieService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+
+@Component
+public class CleanupService {
+  private static final String GROUP = "cleanup";
+
+  @Autowired(required = false)
+  SwabbieService swabbieService;
```

Comment:
Like `Optional<SwabbieService>` autowired in the constructor like you have it in the other class.

Reply:
I have tried to make a component optional based on if a bean was wired in by spring but sadly that hasn't worked for me. `@ConditionalOnBean(....)` or something.

Reply:
And then in controller, would we `Optional<CleanupService>` and then `Assert.notNull` in the controller methods?

Reply:
@robzienert @jeyrschabu what are the advantages of doing that vs what I'm already doing? What I have works when SwabbieService is not enabled, and also when it is. This is controlled in gate config with `@ConditionalOnProperty`

Reply:
@emjburns the `val service = Optional<SwabbieService>` ? mainly because you can do a `service.isPresent()` and log a message indicating the service is not enabled. Vs an exception. More of a preference than anything. What you got will def work.